storage/innobase/handler/ha_innodb.cc

```diff
@@ -18921,12 +18921,15 @@ wsrep_abort_transaction(
 	      wsrep_thd_transaction_state_str(victim_thd),
 	      wsrep_thd_query(victim_thd),
 	      victim_trx ? victim_trx->id : 0);
-
+  wsrep_thd_UNLOCK(victim_thd);
```

Comment:
You changed order of mutexing. How this is going to work if you review normal InnoDB record lock conflict handling where there is no way we could take THD lock first. After you release THD lock that thread could abort, commit or rollback as we do not here have InnoDB mutexes to protect from it.